pennylane/interfaces/jax_jit.py

```diff
@@ -73,10 +73,15 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
     if max_diff > 1:
         raise InterfaceUnsupportedError("The JAX interface only supports first order derivatives.")
 
-    for tape in tapes:
-        # set the trainable parameters
-        params = tape.get_parameters(trainable_only=False)
-        tape.trainable_params = qml.math.get_trainable_indices(params)
+    if _n == 1:

```

Comment:
So if the order of the derivative is higher, we do not need to update the trainable parameters? Why is this? :)

Reply:
You always start at _n=1 and only need to fix your trainable parameters there. For hessian for example you will finish at `_n=2` with only jax arrray and nothing is considered trainable and cannot get the jacobian, therefore I needed to add this condition.